src/emitters/directional.cpp

```diff
@@ -51,7 +51,7 @@ class DirectionalEmitter final : public Emitter<Float, Spectrum> {
                 Throw("Only one of the parameters 'direction' and 'to_world' "
                       "can be specified at the same time!'");
 
-            ScalarVector3f direction(normalize(props.vector3f("direction")));
+            ScalarVector3f direction(normalize(props.array3f("direction")));
```

Comment:
I would prefer it if we still have a thin wrapper `vector3f` and `point3f` that perform a cast from the `Array3f` being stored.

Reply:
Fetching an `array3f` value from the `Properties` object just looks a little strange.